### PR TITLE
calculate build success and failure rate

### LINKED ISSUE #136: KPI Request : Change in success/failure rate
As an OpenShift product manager, I would like to see the change in success or failure rate, so that I can track and measure the effectiveness and success of builds and deployment. 

**Acceptance criteria**
- [x] small notebook in `notebooks/data-sources/TestGrid/metrics/` that collects this metric and stores it in ceph as a parquet.
- [x] Calculate build/deployment success and failure rate
- [ ] Calculate the change in success and failure rate over time

### PR DESCRIPTION
**Pull request by oindrillac**
## Related Issues and Dependencies
Made modifications `build_pass_failure` notebook to calculate build success and failure rate by using "overall" as a proxy for build
Resolves #136 
Related to discussion in #114 

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: notebooks/data-sources/TestGrid/metrics/build_pass_failure.ipynb [hunk 1]
```diff
@@ -2,25 +2,25 @@
  "cells": [
```
Comment (MichaelClifford) [inline notebooks/data-sources/TestGrid/metrics/build_pass_failure.ipynb:2]: <p>Do you still need to limit this number? Won't only looking at "overall" tests drop the number of rows to a more manageable number?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/172/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (MichaelClifford) [inline notebooks/data-sources/TestGrid/metrics/build_pass_failure.ipynb:2]: <ul><li>"Total number of tests:" should this be total number of "builds"?</li></ul><p> </p><ul><li>failing build % = 42.5</li></ul><p>			passing build % = 41.1</p><p>			What are the other 16% of builds doing? Does it have something to do with managing two DF's <code>build_pass</code>    and 		<code>build_fail</code>  . Can't all these values get calculated based only on the build passing df? Shouldn't these two df's be identical except for opposite values in their boolean column at the end?</p><p> </p><ul><li>no_tests = build_passing_df.head(1000000).test.count() = 32,734 so, limiting the DF to the first million rows is not actually doing any work here, you should remove it throughout. </li></ul><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/172/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (MichaelClifford) [inline notebooks/data-sources/TestGrid/metrics/build_pass_failure.ipynb:2]: <p>What does it mean if a build has a value of "False" in both pass and fail column? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/172/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (oindrillac) [reply notebooks/data-sources/TestGrid/metrics/build_pass_failure.ipynb:2]: <p>Yes, we can remove the limit here </p>

Comment (oindrillac) [reply notebooks/data-sources/TestGrid/metrics/build_pass_failure.ipynb:2]: <p>The label value can take one of 16 values in https://github.com/aicoe-aiops/ocp-ci-analysis/blob/1a3a7a58a340c401f8c68ef5d3e9c3f8e11ef864/docs/cell_labels_in_testgrid.md</p><p> </p><p>Although Pass and Fail labels are more common, there could be occurences of Running for the Overall Label, therefore it probably makes sense to calculate "Passing" and "Failing" labels separately.</p><p> </p><p>yes, can remove the (100000) limit everywhere :thumbs_up:</p>

Comment (oindrillac) [reply notebooks/data-sources/TestGrid/metrics/build_pass_failure.ipynb:2]: <p>It means a build contains a value in any of the other labels here: https://github.com/aicoe-aiops/ocp-ci-analysis/blob/1a3a7a58a340c401f8c68ef5d3e9c3f8e11ef864/docs/cell_labels_in_testgrid.md</p><p> </p><p>Although we never see occurences of most of the labels, 1, 4, 12, 13 are labels which are captured, so a false in both means its probably 4 or 13</p>

Comment (oindrillac) [reply notebooks/data-sources/TestGrid/metrics/build_pass_failure.ipynb:2]: <p>Made the requested changes</p>

**Review by MichaelClifford — APPROVED** [review]

**Pull request merged**